File: how_many_numbers_are_smaller_than_the_current_number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        ctr = Counter(nums)
        # print(ctr)    
        
        result = [0] * len(nums)
        
        for num_index in range(len(nums)):
            for key in ctr.keys():
                if key < nums[num_index]:
                    result[num_index] += ctr[key]
        
        return result
        

File: how_many_numbers_are_smaller_than_the_current_number.java
import java.util.Arrays;
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        int [] arr=new int[nums.length];
        for (int i=0; i<nums.length; i++){
            for (int j=0;j<nums.length; j++){
                if (j==i) continue;
                if(nums[j]<nums[i]){
                    arr[i]++;
                }
            }
        }
        return arr;
    }
}
